assets/src/design-system/components/checkbox/index.js

```diff
@@ -62,23 +66,34 @@ const CheckboxContainer = styled.div(
       margin: 0;
       padding: 0;
       opacity: 0;
+
+      :disabled {
+        ~ ${Border} {
+          border-color: ${theme.colors.border.disable};
+        }
+
+        ~ ${StyledCheckmark} {
+          color: ${theme.colors.fg.disable};
+        }
+      }
+
+      &.${FOCUS_VISIBLE_SELECTOR}:not(:active) ~ ${Border} {
+        ${focusCSS(theme.colors.border.focus)};
+      }
+
+      :active ~ ${Border} {
+        border-color: ${theme.colors.border.defaultNormal};
+        box-shadow: 0 0 0 8px ${theme.colors.shadow.active};
+      }
```

Comment:
I love what you're doing here but I'm curious if there's a way to make this into the old theme helper to give a single source of truth to the focus override?  Or at the very least make sure that box shadow is coming from a shared resource. 

My thought here is that there's going to be other components that need a focus style and if it's not centralized like other shared styles are other developers might miss it and create their own in some other capacity and perhaps it affects spacing differently or they add a different color.